src/queue/pgbossQueueProvider.ts

```diff
@@ -0,0 +1,52 @@
+import { Logger } from '@map-colonies/js-logger';
+import PgBoss from 'pg-boss';
+import { inject, injectable } from 'tsyringe';
+import { SERVICES } from '../common/constants';
+import { JOBS_CONFIG_SYMBOL } from '../manager/constants';
+import { JobConfig } from '../manager/interfaces';
+import { QueueProvider, QueueStat } from './queueProvider';
+
+@injectable()
+export class PgBossQueueProvider implements QueueProvider {
+  private readonly queuesNames: Set<string>;
```

Comment:
`queuesNames` and `states` can be combined into the same property

Reply:
as we talked, it cant